datafusion/common/src/dfschema.rs

```diff
@@ -106,37 +106,175 @@ pub type DFSchemaRef = Arc<DFSchema>;
 /// ```
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub struct DFSchema {
+    inner: QualifiedSchema,
+    /// Stores functional dependencies in the schema.
+    functional_dependencies: FunctionalDependencies,
+    /// metadata columns
```

Comment:
Can you provide more documentation  here to document what these are (perhaps adding a link to the higher level description you write on `TableProvider::metadata_columns`)